router/batchrouter/asyncdestinationmanager/salesforce-bulk/types.go

```diff
@@ -0,0 +1,81 @@
+package salesforcebulk
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	oauthv2 "github.com/rudderlabs/rudder-server/services/oauth/v2"
+)
+
+type DestinationConfig struct {
+	RudderAccountID string `json:"rudderAccountId"`
+	Operation       string `json:"operation"`
+	ObjectType      string `json:"objectType"`
+	APIVersion      string `json:"apiVersion"`
+}
+
+type SalesforceBulkUploader struct {
+	destName        string
+	logger          logger.Logger
+	statsFactory    stats.Stats
+	apiService      SalesforceAPIServiceInterface
+	dataHashToJobID map[string][]int64
+	hashMapMutex    sync.RWMutex
+	destinationInfo *oauthv2.DestinationInfo
+}
+
+type SalesforceAPIServiceInterface interface {
+	CreateJob(objectName, operation, externalIDField string) (string, *APIError)
+	UploadData(jobID, csvFilePath string) *APIError
+	CloseJob(jobID string) *APIError
+	GetJobStatus(jobID string) (*JobResponse, *APIError)
+	GetFailedRecords(jobID string) ([]map[string]string, *APIError)
+	GetSuccessfulRecords(jobID string) ([]map[string]string, *APIError)
+	DeleteJob(jobID string) *APIError
+}
+
+type SalesforceAPIService struct {
```

Comment:
Do we need to export it, since it's implementation details?
```suggestion
type apiService struct {
```